From 458b6288bf260212f89bb75ade746a32cb19970a Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Tue, 22 May 2018 13:52:18 +0530
Subject: [PATCH] gengiotypefuncs.py: Read and parse files in binary mode

Fixes this build error on macOS when inside an ssh terminal:

Traceback (most recent call last):
  File "[...]/gio/tests/gengiotypefuncs.py", line 23, in <module>
    for line in f:
  File "[...]/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 2625: ordinal not in range(128)

https://bugzilla.gnome.org/show_bug.cgi?id=796328
---
 gio/tests/gengiotypefuncs.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gio/tests/gengiotypefuncs.py b/gio/tests/gengiotypefuncs.py
index 9732d7892..47219f357 100644
--- a/gio/tests/gengiotypefuncs.py
+++ b/gio/tests/gengiotypefuncs.py
@@ -19,11 +19,11 @@ if debug: print (len(in_files), 'input files')
 
 for filename in in_files:
   if debug: print ('Input file: ', filename)
-  with open(filename, "r") as f:
+  with open(filename, 'rb') as f:
     for line in f:
-      line = line.rstrip('\n').rstrip('\r')
+      line = line.rstrip(b'\n').rstrip(b'\r')
       # print line
-      match = re.search(r'\bg_[a-zA-Z0-9_]*_get_type\b', line)
+      match = re.search(b'\bg_[a-zA-Z0-9_]*_get_type\b', line)
       if match:
         func = match.group(0)
         if not func in funcs:
-- 
2.17.0

